### PR TITLE
Patch for virsh nodeinfo output for special numa arch

### PR DESCRIPTION
**Pull request by jinqi7**
virsh nodeinfo output didn't match Core(s) per socket from host OS for
 special host  which have multiple numa nodes per physical socket, such
 as AMD Bulldozer 62XX familly

Signed-off-by: qijing <jinqi@redhat.com>

### CODE REVIEW

**Comment by jinqi7** [comment]
@waynesun09 Can you help to review the patch?

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py [hunk 1]
```diff
@@ -119,20 +119,36 @@ def output_check(nodeinfo_output):
         cmd = "lscpu | grep 'Core(s) per socket' | head -n1 | awk '{print $4}'"
         cmd_result = utils.run(cmd, ignore_status=True)
         cores_per_socket_os = cmd_result.stdout.strip()
+        spec_numa = False
         if not re.match(cores_per_socket_nodeinfo, cores_per_socket_os):
-            raise error.TestFail("Virsh nodeinfo output didn't match Core(s) "
-                                 "per socket of host OS")
+            # for spec NUMA arch, the output of nodeinfo is in a spec format
+            cmd = "lscpu | grep 'CPU(s)' | head -n1 | awk '{print $2}'"
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: `get_cpu_info()` in utils_misc can help here.

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py [hunk 2]
```diff
@@ -119,20 +119,36 @@ def output_check(nodeinfo_output):
...
+            cmd_result = utils.run(cmd, ignore_status=True)
+            cpus_os = cmd_result.stdout.strip()
+            numa_cells_nodeinfo = _check_nodeinfo(
+                nodeinfo_output, 'NUMA cell(s)', 3)
+            if re.match(cores_per_socket_nodeinfo, cpus_os) and \
+               re.match(numa_cells_nodeinfo, "1"):
+                spec_numa = True
+            else:
+                raise error.TestFail("Virsh nodeinfo output didn't match "
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: please use `test.fail()` here and elsewhere

**Comment by jinqi7** [comment]
@ balamuruhans  Code was changed as discribed in your comment. Please help to review again.

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py [hunk 3]
```diff
@@ -119,20 +119,34 @@ def output_check(nodeinfo_output):
...
-                                 "per socket of host OS")
+            # for spec NUMA arch, the output of nodeinfo is in a spec format
+            cpus_os = utils_misc.get_cpu_info().get("CPU(s)")
+            numa_cells_nodeinfo = _check_nodeinfo(
+                nodeinfo_output, 'NUMA cell(s)', 3)
+            if re.match(cores_per_socket_nodeinfo, cpus_os) and \
+               re.match(numa_cells_nodeinfo, "1"):
+                spec_numa = True
+            else:
+                raise test.fail("Virsh nodeinfo output didn't match "
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: `raise` is not required, can be removed.

**Comment by jinqi7** [comment]
Please help to review again. Thanks

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py [hunk 4]
```diff
@@ -119,20 +119,34 @@ def output_check(nodeinfo_output):
...
         cores_per_socket_os = cmd_result.stdout.strip()
+        spec_numa = False
         if not re.match(cores_per_socket_nodeinfo, cores_per_socket_os):
-            raise error.TestFail("Virsh nodeinfo output didn't match Core(s) "
-                                 "per socket of host OS")
+            # for spec NUMA arch, the output of nodeinfo is in a spec format
+            cpus_os = utils_misc.get_cpu_info().get("CPU(s)")
+            numa_cells_nodeinfo = _check_nodeinfo(
+                nodeinfo_output, 'NUMA cell(s)', 3)
+            if re.match(cores_per_socket_nodeinfo, cpus_os) and \
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: we don't use `\` to do line feed in python

**Comment by will-Do** [comment]
@jinqi7 
Could you please rewrite the patch summary, just declare what the patch done for which case, such as: "virsh_nodeinfo: fix xxxx"

File: libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py (hunk 4 above)
Comment (jinqi7) [inline libvirt/tests/src/virsh_cmd/host/virsh_nodeinfo.py]: @will-Do The "\" is used to do line feed in the same file before, shall I change it also? Thanks.

**Comment by jinqi7** [comment]
@will-Do  Changes are done. Can you pease help to review.? Thanks

**Comment by will-Do** [comment]
Thanks
Next time you can also update the Pull Request subject and summary when the commit subject changed.

**Pull request merged**